syntax = "proto3";

package filetransfer;

service FileTransferService {
  rpc SendFile(stream FileChunk) returns (FileResponse);
  rpc ReceiveFile(FileRequest) returns (stream FileChunk);
  rpc ListFiles(Empty) returns (FileList);
  rpc DeleteFile(FileRequest) returns (FileResponse);
}

message FileRequest {
  string filename = 1;
}

message FileResponse {
  bool success = 1;
  string message = 2;
  int64 file_size = 3;
}

message FileChunk {
  string filename = 1;
  bytes data = 2;
  int64 offset = 3;
  bool is_last = 4;
  int64 total_size = 5;
}

message FileInfo {
  string filename = 1;
  int64 size = 2;
  int64 modified_time = 3;
}

message FileList {
  repeated FileInfo files = 1;
}

message Empty {}

